pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -325,7 +325,7 @@ func newPodSpecForCR(instance *dynatracev1alpha1.OneAgent) corev1.PodSpec {
 			ReadinessProbe: &corev1.Probe{
 				Handler: corev1.Handler{
 					Exec: &corev1.ExecAction{
-						Command: []string{"pgrep", "-f", "oneagentwatchdog"},
+						Command: []string{"ps", "-C", "oneagentwatchdog"},
```

Comment:
Did we ensure the OneAgent docker image contains `ps` and is able to handle the `-C` properly?

By properly I mean, the `-C` used to limit to 15 chars, and the command is 16 chars long:
>        -C cmdlist
>              Select by command name.  This selects the processes whose
>              executable name is given in cmdlist.  NOTE: The command name
>              is not the same as the command line. Previous versions of
>              procps and the kernel truncated this command name to 15
>              characters. This limitation is no longer present in both. If
>              you depended on matching only 15 characters, you may no longer
>              get a match.
http://man7.org/linux/man-pages/man1/ps.1.html

Reply:
Typically, it should be given as `ps -C oneagentwatchdo`

Reply:
Imo, we should add `apt-get update && apt-get install procps` while creating docker container for the same, there are several instances where ps does not work without explicit installation. 

Second, I'm still trying to see the advantage of using `ps` over `pgrep` - because if we're only concerned with the executable, then `pgrep <executable>` should work okay.

Reply:
To handle installation of procps, we could add it to - `docker.io/dynatrace/oneagent:latest` perhaps?